src/tables.ts

```diff
@@ -100,3 +100,18 @@ export const userstatshistory = new DatabaseTable<{
 	usercount: number;
 	programid: 'showdown' | 'po';
 }>(psdb, 'userstatshistory', 'id');
+
+// oauth stuff
+
+export const oauthClients = new DatabaseTable<{
+	owner: string; // ps username
+	client_title: string;
+	id: string; // hex hash
+}>(psdb, 'oauth_clients', 'id');
+
+export const oauthTokens = new DatabaseTable<{
+	owner: string;
+	client: string; // id of client
```

Comment:
Is this the ID or the title of the client? In the SQL, it's a varchar(40), just like the title, while the ID is a varchar(32). I suppose you could just leave 8 characters unused, but let's make this clearer.

Reply:
ID.

Reply:
OK, so why does it need to be a `varchar(40)` in the schema when the client ID is only 32 characters?

Reply:
doesn’t need to be, I’ll swap it later. just had it like this because I picked an arbitrary number.